soco/discovery.py

```diff
@@ -215,27 +243,180 @@ def any_soco():
+
+        Args:
+            min_netmask (int): The minimum number of netmask bits. Used to
+                constrain the network search space.
+
+        Returns:
+            set: A set of attached networks.
+        """
+        ipv4_net_list = set()
+        adapters = ifaddr.get_adapters()
+        for adapter in adapters:
+            for ip_address in adapter.ips:
+                if is_ipv4_address(ip_address.ip):
+                    network_ip = ipaddress.ip_network(ip_address.ip)
+                    # Restrict to private networks and exclude loopback
+                    if network_ip.is_private and not network_ip.is_loopback:
+                        # Constrain the size of network that will be searched
+                        if ip_address.network_prefix < min_netmask:
+                            ip_address.network_prefix = min_netmask
+                        network = ipaddress.ip_network(
+                            ip_address.ip + "/" + str(ip_address.network_prefix), False
+                        )
+                        ipv4_net_list.add(network)
+        _LOG.info("List of networks to search: %s", str(ipv4_net_list))
+        return ipv4_net_list
+
+    def check_ip_and_port(ip_address, port, timeout):
+        """Helper function to check if a port is open"""
+        _socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        _socket.settimeout(timeout)
+        return not bool(_socket.connect_ex((ip_address, port)))
+
+    def is_sonos(ip_address):
+        """Helper function to check if this is a Sonos device"""
+        try:
+            # Try getting a device property
+            _ = config.SOCO_CLASS(ip_address).is_visible
+            return True
+        # The exception is unimportant
+        # pylint: disable=bare-except
+        except:  # noqa: E722
+            return False
+
+    def sonos_scan_worker_thread(ip_list, socket_timeout, sonos_ip_addresses):
+        """Helper function worker thread to take IP addresses off a list and
+        test whether there is (1) a device with port 1400 open at that IP
+        address, then (2) check the device is a Sonos device.
+        Once a there is a hit, the list is cleared to prevent any further
+        checking of addresses by any thread.
+        """
+        while len(ip_list) > 0:
+            ip_address = str(ip_list.pop())
+            if check_ip_and_port(ip_address, 1400, socket_timeout):
+                _LOG.info("Found open port 1400 at IP '%s'", ip_address)
+                if is_sonos(ip_address):
+                    _LOG.info("Confirmed Sonos device at IP '%s'", ip_address)
+                    sonos_ip_addresses.append(ip_address)
+                    # Clear the list to eliminate further searching by
+                    # all threads
+                    ip_list.clear()
+
+    # Generate the set of IPs to check
+    ip_set = set()
+    for network in find_ipv4_networks(min_netmask):
+        for ip_address in network:
```

Comment:
If network is iterable, it might be possible to get rid of the inner loop with something like:

```python
for network in find_ipv4_networks(min_netmaks):
    ip_set.update(set(network))
```

Reply:
Very neat. The inner `set()` cast is not required as `network` is already a set. Implemented in https://github.com/SoCo/SoCo/commit/5e4465ffa3791d61f51c56326c71a0fc96342afe.

Reply:
Oh yeah, but a set that you create or one that is provided by the library. The type of return values from the library could change but "being iterable" is much less likely to change.